contracts/FunctionsConsumer.sol

```diff
@@ -1,86 +1,80 @@
+import {FunctionsClient} from "./@chainlink/contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol";
+import {ConfirmedOwner} from "./@chainlink/contracts/src/v0.8/shared/access/ConfirmedOwner.sol";
+import {FunctionsRequest} from "./@chainlink/contracts/src/v0.8/functions/dev/1_0_0/libraries/FunctionsRequest.sol";
 
 /**
- * @title Functions Consumer contract
- * @notice This contract is a demonstration of using Functions.
- * @notice NOT FOR PRODUCTION USE
+ * @title Chainlink Functions example on-demand consumer contract example
  */
 contract FunctionsConsumer is FunctionsClient, ConfirmedOwner {
-  using Functions for Functions.Request;
+  using FunctionsRequest for FunctionsRequest.Request;
 
-  bytes32 public latestRequestId;
-  bytes public latestResponse;
-  bytes public latestError;
+  bytes32 public donId; // DON ID for the Functions DON to which the requests are sent
 
-  event OCRResponse(bytes32 indexed requestId, bytes result, bytes err);
+  bytes32 public s_lastRequestId;
+  bytes public s_lastResponse;
+  bytes public s_lastError;
+
+  error UnexpectedRequestID(bytes32 requestId);
+
+  constructor(address router, bytes32 _donId) FunctionsClient(router) ConfirmedOwner(msg.sender) {
+    donId = _donId;
+  }
 
   /**
-   * @notice Executes once when a contract is created to initialize state variables
-   *
-   * @param oracle - The FunctionsOracle contract
+   * @notice Set the DON ID
+   * @param newDonId New DON ID
    */
-  // https://github.com/protofire/solhint/issues/242
-  // solhint-disable-next-line no-empty-blocks
-  constructor(address oracle) FunctionsClient(oracle) ConfirmedOwner(msg.sender) {}
+  function setDonId(bytes32 newDonId) external onlyOwner {
+    donId = newDonId;
+  }
 
   /**
-   * @notice Send a simple request
-   *
+   * @notice Triggers an on-demand Functions request using remote encrypted secrets
    * @param source JavaScript source code
-   * @param secrets Encrypted secrets payload
-   * @param args List of arguments accessible from within the source code
-   * @param subscriptionId Funtions billing subscription ID
-   * @param gasLimit Maximum amount of gas used to call the client contract's `handleOracleFulfillment` function
-   * @return Functions request ID
+   * @param secretsLocation Location of secrets (only Location.Remote & Location.DONHosted are supported)
+   * @param encryptedSecretsReference Reference pointing to encrypted secrets
+   * @param args String arguments that will be passed into the source code
+   * @param bytesArgs Bytes arguments that will be passed into the source code as hex strings
+   * @param subscriptionId Subscription ID used to pay for request (FunctionsConsumer contract address must first be added to the subscription)
+   * @param callbackGasLimit Maximum amount of gas used to call the client contract's `handleOracleFulfillment` method
    */
-  function executeRequest(
+  function sendRequest(
```

Comment:
@KuphJr 
to address your point that people dont realise that this example contract is not exhaustive, we could have another func here called sendRequestAsCBOR() to illustrate the off chain cbor encoding of requests

Reply:
My sense is that fewer people will start with AutomatedFunctionsConsumer as it is more advanced and requires more config (higher time to satisfied feeling)

This contract will be viewed and interacted with earlier in the user journey most of the time, and also may be demo'd more.  I recommend it be here as well.  

Once request.js is done I can base a different CLI task off of that to run CBOR encoded requests if you like

Reply:
Yeah.  If you feel strongly that you want an on-demand example that uses an off-chain encoded CBOR request, I would prefer to keep that in a separate example to keep `request.js` as straightforward as possible.

We can revisit once we complete the currently scoped work for the starter kit.